webapp/channels/src/components/widgets/tag/generic_tag.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import React from 'react';
+
+import './generic_tag.scss';
+
+export type GenericTagProps = {
+
+    /**
+     * The text content of the tag
+     */
+    text: string;
+
+    /**
+     * Optional CSS class name for custom styling
+     */
+    className?: string;
+
+    /**
+     * Optional click handler
+     */
+    onClick?: (e: React.MouseEvent) => void;
+
+    /**
+     * Optional variant for different visual styles
+     */
+    variant?: 'default' | 'primary' | 'secondary' | 'info';
+
+    /**
+     * Optional size
+     */
+    size?: 'small' | 'medium' | 'large';
+
+    /**
+     * Optional test ID for automated testing
+     */
+    testId?: string;
+};
+
+/**
+ * A generic tag component that can be used to display labeled information
+ */
+const GenericTag: React.FC<GenericTagProps> = ({
```

Comment:
Should we add a tooltip property as well? We can use `components/with_tooltip` to render the tooltip. It was in the spec to display property name.

Reply:
hey, yes, of course, I missed that detail. I will add it in an upcoming PR.